contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol

```diff
@@ -718,6 +723,37 @@ contract OCR2DRRegistry is ConfirmedOwner, OCR2DRRegistryInterface, ERC677Receiv
     return false;
   }
 
+  /**
+   * @notice Total the estimated cost of all pending request commitments for all consumers for a given sub.
+   * @param subscriptionId - ID of the subscription
+   * @return costEstimate estimated cost in Juels of LINK that will be charged once all pending requests are completed
+   * @dev Looping is bounded to MAX_CONSUMERS*(number of DONs).
+   * @dev TODO: use pointer to only iterate possibly pending requests
+   */
+
+  function estimateCostPendingRequests(uint64 subscriptionId) public view returns (uint96) {
+    address[] memory consumers = s_subscriptionConfigs[subscriptionId].consumers;
+    address[] memory authorizedSendersList = getAuthorizedSenders();
+    uint96 cost;
+    for (uint256 i = 0; i < consumers.length; i++) {
+      for (uint256 j = 0; j < authorizedSendersList.length; j++) {
+        bytes32 requestId = computeRequestId(
+          authorizedSendersList[j],
+          consumers[i],
+          subscriptionId,
+          s_consumers[consumers[i]][subscriptionId]
+        );
+        cost += estimateCost(
```

Comment:
A simpler approach would be to keep track of cumulative "blockedBalance" - add on startBilling, subtract on fulfill or timeout. It's not as good as this implementation if gas cost changes quickly but maybe good enough?

Reply:
I like it